src/udx.c

```diff
@@ -407,7 +419,7 @@ process_packet (udx_t *self, char *buf, ssize_t buf_len) {
     process_sacks(stream, buf, buf_len);
   }
 
-  if (type & UDX_HEADER_DATA_OR_END && udx__cirbuf_get(inc, seq) == NULL && (stream->status & UDX_STREAM_SHOULD_READ) == UDX_STREAM_READ) {
+  if (type & UDX_HEADER_DATA_OR_END && seq_compare(stream->ack, seq) <= 0 && udx__cirbuf_get(inc, seq) == NULL && (stream->status & UDX_STREAM_SHOULD_READ) == UDX_STREAM_READ) {
```

Comment:
This checks if `seq` is either the next or a future packet. Is that correct?

Reply:
yes!